tasks/human_eval_007.rs

```diff
@@ -9,7 +9,101 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+fn filter_by_substring<'a>(strings: &Vec<&'a str>, substring: &str) -> (res: Vec<&'a str>)
+    ensures
```

Comment:
I've rewritten the the spec to capture the prompt

Reply:
I think this version captures the property I mentioned above, but I think it might have lost the property your previous version included.  In other words, I think your current spec would allow the function to return the entire input list.

Perhaps you could show that combining your two versions of the spec imply the `filter`-based spec above.